src/Microsoft.AspNet.Http/HeaderDictionary.cs

```diff
@@ -1,199 +0,0 @@
-
-using System;
-using System.Collections;
-using System.Collections.Generic;
-using Microsoft.Framework.Primitives;
-
-namespace Microsoft.AspNet.Http.Internal
-{
-    /// <summary>
-    /// Represents a wrapper for owin.RequestHeaders and owin.ResponseHeaders.
-    /// </summary>
-    public class HeaderDictionary : IHeaderDictionary
-    {
-        public HeaderDictionary() : this(new Dictionary<string, StringValues>(StringComparer.OrdinalIgnoreCase))
-        {
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="T:Microsoft.Owin.HeaderDictionary" /> class.
-        /// </summary>
-        /// <param name="store">The underlying data store.</param>
-        public HeaderDictionary(IDictionary<string, StringValues> store)
-        {
-            if (store == null)
-            {
-                throw new ArgumentNullException(nameof(store));
-            }
-
-            Store = store;
-        }
-
-        private IDictionary<string, StringValues> Store { get; set; }
-
-        /// <summary>
-        /// Gets an <see cref="T:System.Collections.ICollection" /> that contains the keys in the <see cref="T:Microsoft.Owin.HeaderDictionary" />;.
-        /// </summary>
-        /// <returns>An <see cref="T:System.Collections.ICollection" /> that contains the keys in the <see cref="T:Microsoft.Owin.HeaderDictionary" />.</returns>
-        public ICollection<string> Keys
-        {
-            get { return Store.Keys; }
-        }
-
-        /// <summary>
-        /// 
-        /// </summary>
-        public ICollection<StringValues> Values
-        {
-            get { return Store.Values; }
-        }
-
-        /// <summary>
-        /// Gets the number of elements contained in the <see cref="T:Microsoft.Owin.HeaderDictionary" />;.
-        /// </summary>
-        /// <returns>The number of elements contained in the <see cref="T:Microsoft.Owin.HeaderDictionary" />.</returns>
-        public int Count
-        {
-            get { return Store.Count; }
-        }
-
-        /// <summary>
-        /// Gets a value that indicates whether the <see cref="T:Microsoft.Owin.HeaderDictionary" /> is in read-only mode.
-        /// </summary>
-        /// <returns>true if the <see cref="T:Microsoft.Owin.HeaderDictionary" /> is in read-only mode; otherwise, false.</returns>
-        public bool IsReadOnly
-        {
-            get { return Store.IsReadOnly; }
-        }
-
-        /// <summary>
-        /// Get or sets the associated value from the collection as a single string.
-        /// </summary>
-        /// <param name="key">The header name.</param>
-        /// <returns>the associated value from the collection as a StringValues or StringValues.Empty if the key is not present.</returns>
-        public StringValues this[string key]
-        {
-            get { return ParsingHelpers.GetHeader(Store, key); }
```

Comment:
Where did this logic go? Seems like none of the tests failed. Which either means we have no coverage or this is equivalent to looking up the key in the dictionary.